test/module/irohad/consensus/yac/network_test.cpp

```diff
@@ -18,52 +18,95 @@
 #include "module/irohad/consensus/yac/yac_mocks.hpp"
 
 #include <grpc++/grpc++.h>
-#include "consensus/yac/transport/impl/network_impl.hpp"
 
-using namespace iroha::consensus::yac;
-using iroha::model::Peer;
+#include "consensus/yac/transport/yac_pb_converters.hpp"
+#include "consensus/yac/transport/impl/network_impl.hpp"
 
 using ::testing::_;
 using ::testing::InvokeWithoutArgs;
 
-TEST(NetworkTest, MessageHandledWhenMessageSent) {
-  auto notifications = std::make_shared<MockYacNetworkNotifications>();
+namespace iroha {
+  namespace consensus {
+    namespace yac {
+      class YacNetworkTest : public ::testing::Test {
+       public:
+        void SetUp() override {
+          notifications = std::make_shared<MockYacNetworkNotifications>();
+
+          peer = mk_peer("0.0.0.0:50051");
+          network = std::make_shared<NetworkImpl>();
+
+          message.hash.proposal_hash = "proposal";
+          message.hash.block_hash = "block";
+
+          network->subscribe(notifications);
+
+          grpc::ServerBuilder builder;
+          int port = 0;
+          builder.AddListeningPort(peer.address,
+                                   grpc::InsecureServerCredentials(),
+                                   &port);
+          builder.RegisterService(network.get());
+          server = builder.BuildAndStart();
+          ASSERT_TRUE(server);
+          ASSERT_NE(port, 0);
+        }
 
-  auto peer = mk_peer("0.0.0.0:50051");
-  std::vector<Peer> peers = {peer};
-  std::shared_ptr<NetworkImpl> network =
-      std::make_shared<NetworkImpl>(peer.address, peers);
+        void TearDown() override {
+          server->Shutdown();
+        }
 
-  VoteMessage message;
-  message.hash.proposal_hash = "proposal";
-  message.hash.block_hash = "block";
+        std::shared_ptr<MockYacNetworkNotifications> notifications;
+        std::shared_ptr<NetworkImpl> network;
+        model::Peer peer;
+        VoteMessage message;
+        std::unique_ptr<grpc::Server> server;
+        std::mutex mtx;
+        std::condition_variable cv;
+      };
 
-  std::mutex mtx;
-  std::condition_variable cv;
-  ON_CALL(*notifications, on_vote(peer, message))
-      .WillByDefault(
-          InvokeWithoutArgs(&cv, &std::condition_variable::notify_one));
+      /**
+       * @given initialized network
```

Comment:
Pls, make a rich description of the test scenario.

Reply:
Simple test = simple description, don't see any problem here